### ISSUE TITLE
make strings that look like modules clickable

### ISSUE THREAD

**Issue by KotlinIsland** [DetachHead/basedpyright#313]
```py
"collections"   # should be clickable
```

**Comment by DetachHead** [comment]
i feel like this could cause false positives and should only do it in certain cases. maybe using a plugin (#22) to support it for specific things

**Comment by DetachHead** [comment]
what about a `ModuleString` type that enforces that the string resolves to a valid module

**Comment by KotlinIsland** [comment]
Would want `SymbolString` as well.

And would it be strict(reject `str`), or just report when an invalid literal is provided?
